src/compositemark.ts

```diff
@@ -0,0 +1,64 @@
+import {Encoding} from './encoding';
```

Comment:
I'm not a fan or premature abstraction. In this case we have an abstraction for a single instance! Why?